opentreemap/treemap/udf.py

```diff
@@ -1,29 +1,86 @@
 # -*- coding: utf-8 -*-
+
+'''
+treemap.udf
+
+Entry points:
+-   `UDFModel`, a base class for models that need user defined fields
+-   `UserDefinedFieldDefinition`, defines the name, data type, and model type
+    for a user defined field, for a given treemap instance, as a
+    scalar or a collection
+-   `UserDefinedCollectionValue`, an HStore collection value for a
+    specific `UserDefinedFieldDefinition` on a specific model instance
+
+`UDFModel` user guide:
+
+Classes that want custom fields subclass `UDFModel`.
+The `UDFModel` `HStoreField` is named `hstore_udfs`, but unless you are
+getting deep into udf code, you should use `udfs`, which is an ordinary
+attribute on the model instance, not directly associated with any
+django `Field`.
+
+To assign a custom field,
+    `my_model_instance.udfs['custom field name'] = value`
+
+There are three ways to retrieve it:
+    `my_model_instance.udfs['custom field name']`
+    `getattr(my_model_instance, 'custom field name')`
+    `getattr(my_model_instance, 'udf:custom field name'`)
+
+All of the above work for both scalar and collection custom fields.
+
+Filter using the `hstore_udf` transform.
+
+In addition to the transforms and lookups described at
+https://docs.djangoproject.com/en/1.8/ref/contrib/postgres/fields/,
+`UDFModel` implements `__int` and `__float` transforms, for use
+before magnitude comparisons (`__gt`, `__gte`, `__lt`, `__lte`).
+
+`udfs` perpetually syncs to `hstore_udfs`.
+Values in `udfs` are always typed for the app, and
+values in `hstore_udfs` are always typed for the db.
+Since they are perpetually clean, code that assigns to `udfs`
+needs to be prepared for a `ValidationError` at the time of the
+assignment, not just at `save`.
+
+In practice, this is not a problem, because the web frontend
+takes care to send valid data in PUTs and POSTs.
```

Comment:
Hmmmm.

It may be necessary to write more tests for both of them to be sure.

I'll write additional tests, and get back to you if it turns out to be a problem.